rmf_site_editor/src/site/scenario.rs

```diff
@@ -102,65 +135,214 @@ pub fn update_current_scenario(
+                            .zip(Some(c_entity))
+                    })
+                else {
+                    continue;
+                };
+                let instance = instance.as_mut();
+
+                match update.update_type {
+                    UpdateInstanceType::Include => {
+                        let instance_pose = instance.pose().or(recall_instance
+                            .get(*scenario_child)
+                            .ok()
+                            .and_then(|r| r.pose));
 
-                let existing_moved_instance = current_scenario
-                    .moved_instances
-                    .iter_mut()
-                    .find(|(e, _)| *e == entity);
-                if let Some(existing_moved_instance) = existing_moved_instance {
-                    existing_moved_instance.1 = pose.clone();
-                    return;
-                } else {
-                    current_scenario
-                        .moved_instances
-                        .push((entity, pose.clone()));
-                    return;
+                        if parent_pose.is_some() {
+                            *instance = Instance::new_inherited(instance_pose)
+                        } else if let Some(instance_pose) = instance_pose {
+                            *instance = Instance::new_added(instance_pose);
+                        } else {
+                            let instance_id = model_instances
+                                .get(update.instance)
+                                .map(|n| n.0.clone())
+                                .unwrap_or(format!("{}", update.instance.index()).to_string());
+                            error!(
+                                "Unable to retrieve pose for instance {:?}, \
+                                setting to default pose as AddedInstance in current scenario",
+                                instance_id
+                            );
+                            *instance = Instance::new_added(Pose::default());
+                        }
+                    }
+                    UpdateInstanceType::Hide => {
+                        *instance = Instance::new_hidden();
+                    }
+                    UpdateInstanceType::Modify(new_pose) => {
+                        // Update Pose changes
+                        match instance {
+                            Instance::Added(_) => *instance = Instance::new_added(new_pose.clone()),
+                            Instance::Inherited(_) => {
+                                *instance = Instance::new_inherited(Some(new_pose.clone()))
+                            }
+                            _ => {}
+                        }
+                    }
+                    UpdateInstanceType::ResetPose => {
+                        if parent_pose.is_some() {
+                            *instance = Instance::new_inherited(None)
+                        }
+                    }
+                    _ => {}
```

Comment:
In general I suspect it would be better to merge this whole match block with the `UpdateInstanceType::Added` match block that lives outside of this. I think splitting them based on whether you expect to spawn an entity is not sound since you can't **_really_** guarantee whether you really want to spawn an entity or modify a component in any of these cases.

One option that could simplify things is to insert a system which always runs before this one which adds an entity with `Instance::Inherited` and `Affiliated` components every time a new entity is created with a `Scenario` or `InstanceMarker` component. Essentially it would make sure the matrix of modifier entities is always filled in before this system is run.